cloudinit/sources/DataSourceOracle.py

```diff
@@ -297,32 +298,40 @@ def _add_network_config_from_opc_imds(self, set_primary: bool = False):
                 )
                 continue
             name = interfaces_by_mac[mac_address]
-            prefix = ipaddress.ip_network(
-                vnic_dict["subnetCidrBlock"]
-            ).prefixlen
+            network = ipaddress.ip_network(vnic_dict["subnetCidrBlock"])
 
             if self._network_config["version"] == 1:
-                subnet = {
-                    "type": "static",
-                    "address": f"{vnic_dict['privateIp']}/{prefix}",
+                interface_config = {
+                    "name": name,
+                    "type": "physical",
+                    "mac_address": mac_address,
+                    "mtu": MTU,
+                    "subnets": [
+                        {
+                            "address": (
+                                f"{vnic_dict['privateIp']}/{network.prefixlen}"
+                            )
+                        }
+                    ],
                 }
-                self._network_config["config"].append(
-                    {
-                        "name": name,
-                        "type": "physical",
-                        "mac_address": mac_address,
-                        "mtu": MTU,
-                        "subnets": [subnet],
-                    }
+                # Primary interface uses DHCP as is our first entry
+                interface_config["subnets"][0]["type"] = (
+                    "dhcp" if is_primary else "static"
                 )
+                self._network_config["config"].append(interface_config)
             elif self._network_config["version"] == 2:
-                self._network_config["ethernets"][name] = {
-                    "addresses": [f"{vnic_dict['privateIp']}/{prefix}"],
+                # Why does this elif exist???
+                # Are there plans to switch to v2?
```

Comment:
Good question on the elif here. This is precisely the reason. e7881d5c62e68e4962e2fcd3d12089e5a3b94dc9

Generally, yes we want clouds to migrate towards v2 for better expression of complex networks/tags/metrics/policy etc. I think that is a long-term goal for any cloud IMDS to move toward v2 and eventually retire our v1 support.

It was only paving the way to move toward v2, some discussions were had a cloud-init summit to try to move this direction with ORacle's IMDS. It'd be nice to continue that work, but as it stands, I don't think any path can actually get to this currently.